### PR TITLE
ci(workflow): add "lint" for branch name conventions

### PR DESCRIPTION
**Pull request by ccamel**
Let @bot-anik verify that opened PRs follow our naming conventions, and close them if not (the branch is still opened).

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -87,3 +87,37 @@ jobs:
...
           token: ${{ secrets.OKP4_TOKEN }}
+
+  lint-branch-name:
+    runs-on: ubuntu-20.04
+    if: github.actor != 'dependabot[bot]' && github.event_name == 'pull_request' && (github.event.action == 'opened' || github.event.action == 'reopened')
+    steps:
+      - name: Check branch name conventions
+        uses: AlbertHernandez/branch-name-action@v1.0.2
+        with:
+          branch_pattern: "feat|fix|hotfix|docs|ci|chore"
```
Comment (fredericvilcot) [inline .github/workflows/lint.yml:98]: That sounds great!
What about 'style' and 'build' pull requests?

Comment (ccamel) [reply .github/workflows/lint.yml:98]: Yes I'll add it.

**Review by fredericvilcot — DISMISSED** [review]

**Review by ad2ien — APPROVED** [review]

**Review by bdeneux — DISMISSED** [review]

**Review by amimart — DISMISSED** [review]

**Comment by ErikssonJoakim** [comment]
I am having trouble understanding. How is `branch_pattern: "feat|fix|hotfix|docs|ci|chore"` enforcing the branch pattern `<type>/<name>`?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (ccamel) [reply .github/workflows/lint.yml]: Updated the description as well (by https://github.com/okp4/ui/pull/193/commits/e9db80ea1cc751e97f0ff8eb24d0ab622aa7f4d3).

**Review by bdeneux — APPROVED** [review]

**Comment by ccamel** [comment]
@ErikssonJoakim 

> I am having trouble understanding. How is `branch_pattern: "feat|fix|hotfix|docs|ci|chore"` enforcing the branch pattern `<type>/<name>`?

We're using the following GitHub action [AlbertHernandez/branch-name-action](https://github.com/AlbertHernandez/branch-name-action) to check if the branch name is following a convention name. In the documentation, you'll see that the property `branch_pattern` is a "pattern" (a [regular expression](https://en.wikipedia.org/wiki/Regular_expression)) that a branch name should follow (you can see how it is used in the following [code](https://github.com/AlbertHernandez/branch-name-action/blob/master/src/run-action.js#L15)). If you're not really familiar with regular expression, you can play live at this site: https://regex101.com/r/Na6FIe/1.

**Review by fredericvilcot — APPROVED** [review]
Great! Thanks @ccamel!

**Comment by ErikssonJoakim** [comment]
@ccamel 
Shouldn't the pattern enforce all rules from [the naming rules wiki](https://github.com/okp4/wiki/wiki/Naming-rules) in that case? Something like this :
`<type>\/[-?[a-z0-9@$#&*\{\}\[\]\(\),=\-\.;'\+\/]+-?]*`
[regex example here](https://regex101.com/r/Po5XZ9/1)

OBS : Pattern not exhaustive to all rules

**Review by amimart — APPROVED** [review]

**Comment by ccamel** [comment]
@ErikssonJoakim I see your point. We can harden the check a little more, but I don't want to go too strict - we need to strike a proper balance. What I suggest to you is to consider the current proposal as it is and in a second time, consider a stricter approach that we can all discuss together. Is that okay?

**Review by ErikssonJoakim — APPROVED** [review]
@ccamel That works, looking forward to discussing regex patterns further with you😁

**Pull request merged**